### ISSUE TITLE
Unable to create apm policy/profile/policy-items 

### ISSUE THREAD

**Issue by atechnicnate** [F5Networks/f5-ansible#1130]
##### ISSUE TYPE
 - Feature Idea

##### COMPONENT NAME
APM Policy-items, policy and profile creation

##### ANSIBLE VERSION
```
ANY
```
##### PYTHON VERSION
```
ANY
```
##### BIGIP VERSION
```
ANY
```

##### LIBRARY VERSIONS
```
ANY
```

##### CONFIGURATION
Currently this does not work in Ansible as APM requires these items to be created at the same time in a single transaction.

##### OS / ENVIRONMENT
In tmsh this is done with these commands:
tmsh create cli transaction
create apm policy agent ending-allow /Common/eple_end_allow_ag { }
create apm policy agent ending-deny /Common/eple_end_deny_ag { }
create apm policy agent ending-deny /Common/eple_end_deny2_ag { }
create apm policy policy-item /Common/eple_end_allow { agents add { /Common/eple_end_allow_ag { type ending-allow } } caption Allow color 1 item-type ending }
create apm policy policy-item /Common/eple_end_deny { agents add { /Common/eple_end_deny_ag { type ending-deny } } caption Deny color 2 item-type ending }
create apm policy policy-item /Common/eple_end_deny2 { agents add { /Common/eple_end_deny2_ag { type ending-deny } } caption Deny2 color 4 item-type ending }
create apm policy policy-item /Common/eple_ent { caption Start color 1 rules { { caption fallback next-item /Common/eple_end_deny } } }
create apm policy access-policy /Common/eple { default-ending /Common/eple_end_deny items add { eple_end_allow { } eple_end_deny { } eple_end_deny2 { } eple_ent { } } start-item eple_ent }
create apm profile access /Common/eple { accept-languages add { en } access-policy /Common/eple}
submit cli transaction

##### SUMMARY
APM requires policy-items to be created at the same time as the policy/tasks but currently there is no way to do this via Ansible.

##### STEPS TO REPRODUCE
```

```

##### EXPECTED RESULTS
APM specific module for this task

##### ACTUAL RESULTS
```
No module exists to do this and no workaround exists at this time.
```

**Comment by Sam-Hall** [comment]
Is there now a workaround? It seems to me it would be possible to include an un-tgz'd "template" of the access policy export in my Ansible file structure. Then playbook will copy and perhaps process some j2 template files, storing the final access policy file structure in a temp directory before compressing it back to a tgz and importing it. In theory this seems like a viable workaround to me, is there any reason this can't work?

**Comment by Sam-Hall** [comment]
What I'm doing is to extract the access policy tar.gz files into the Ansible directory structure somewhere, then I have a vars file alongside all those directories with md5 checksum dicts which the playbook checks to confirm the working directory is something that's ready for release before uploading it. Similar to what we used to do manually. For example:
```
dirs:
  - "test_ap1"
  - "test_ap2"

releases:
  5cdc9c2e27f8c8b326e87d2c462ca18a:
    name: "test_ap1_v0.2"
    dir: "test_ap1"
    hosts: "dev"
  92753b5e577772ff8d47175639a573c3:
    name: "test_ap1_v0.1"
    dir: "test_ap1"
    hosts: "tst"
  a5cd7d25247ecc73ec330a07f5be2350:
    name: "test_ap2_v0.1"
    dir: "test_ap2"
    hosts: "dev"
```

I'm using shell module with a funky combination of find, sort and md5sum Linux commands to generate a md5sum of the whole directory tree, for want of a better way to md5 a directory. Anyway, it's working well enough. Ansible looks for each access policy in the dirs list and then performs the same md5 check on it to determine if the files in their current state are registered for release, what we want to name it and where it should be deployed. We just have the latest version of the access policy in the directory structure, assuming we can always go back by delving into version control.

I also realised that some access policies contain secrets so I've used Ansible vault to encrypt any such files, then the copy module with the decrypt option to copy the directory structure into a tmp directory before re-tgzing.

Minor changes are still going to be a pain via Ansible. I don't have any use cases for jinja templates in the access policy files yet, but I had an idea about using that for dynamically populating the host list of a multi-domain access policy. That would probably upset the whole release management thing we've got happening now, and I think it might be better to manage the hostname list via tmsh command so we don't have to do a full blown release everytime a new hostname is added (and update all the virtuals to use the new AP). I have an access policy with over 40 sites on it, using the existing Ansible f5 modules to switch them all over after an update of the policy would take an unacceptably long time.
